CONTRIBUTING.md

```diff
@@ -0,0 +1,87 @@
+# Contributing
+
+Thanks for being willing to contribute!
+
+Working on your first Pull Request? You can learn how from this free series
+[How to Contribute to an Open Source Project on GitHub](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github).
+
+## Project setup
+
+1. Fork this repository
+2. Clone your forked repository
+3. Run `npm install` to install corresponding dependencies
+4. Create a branch for your PR named like `pr/your-branch-name` (you can do this through git CLI with `git checkout -b pr/your-branch-name`)
+
+> Tip: Keep your `master` branch pointing at the original repository and make
+> pull requests from branches on your fork. To do this, run:
+>
+> ```
+> git remote add upstream https://github.com/testing-library/eslint-plugin-testing-library.git
+> git fetch upstream
+> git branch --set-upstream-to=upstream/master master
+> ```
+>
+> This will add the original repository as a "remote" called "upstream," Then
+> fetch the git information from that remote, then set your local `master`
+> branch to use the upstream master branch whenever you run `git pull`. Then you
+> can make all of your pull request branches based on this `master` branch.
+> Whenever you want to update your version of `master`, do a regular `git pull`.
+
+## Committing and Pushing changes
+
+There are git hooks config with this project that are automatically installed
+and set up when you install dependencies. This will be run on every commit:
+
+- check all tests are passing, code validation is fine and format files automatically
+- check commit message is following [Conventional Commit specification](https://www.conventionalcommits.org/en/v1.0.0/)
+
+You can run `npm run test:update` which will update any snapshots that need updating.
+
+## Rule naming conventions
+
+Based on [ESLint's Rule Naming Conventions](https://eslint.org/docs/developer-guide/working-with-rules#rule-naming-conventions), this is the simple convention you must follow:
+
+- If your rule is disallowing something, prefix it with `no-` such as `no-debug`
+  for disallowing `debug()`.
+- If your rule is suggesting to use a specific way for something that can be
+  done in several ways, you could **optionally** prefix it with `prefer-` such as
+  `prefer-screen-queries` for suggesting to use `screen.getByText()` from
+  imported `screen` rather than`getByText()` from `render`'s result,
+  even though both are technically fine.
```

Comment:
```suggestion
- If your rule is suggesting to prefer a way of doing something, among other ways, you can **optionally** prefix it with `prefer-`. For example, `prefer-screen-queries` suggests to use `screen.getByText()` from imported `screen` rather than`getByText()` from `render`'s result though both are technically fine.
```